packages/auth/amplify_auth_cognito_test/test/state/configuration_state_machine_test.dart

```diff
@@ -83,5 +86,18 @@ void main() {
 
       await stateMachine.close();
     });
+
+    test('configures analytics metadata if configured', () async {
+      stateMachine
+        ..getOrCreate(ConfigurationStateMachine.type)
+        ..dispatch(ConfigurationEvent.configure(mockConfigWithPinpoint));
+
+      expect(
```

Comment:
Update to await configuration after moving logic to `ConfigurationStateMachine`. This should only be true after configuration completes.

Reply:
EndpointInfoManager probably should be mocked as well.

Reply:
I don't think EndpointInfoManager needs to be mocked, but we can inject a mock secure storage and test with that